File: case_study/Frontend/assetflow/src/services/serviceRequestService.js
// ─── Service Request Service ──────────────────────────────────────────────────
import { api } from "./api";

export const getAllServiceRequests = (token) =>
  api("GET", "/service-requests", null, token);

export const getServiceRequestsByEmployee = (userId, token) =>
  api("GET", `/service-requests/employee/${userId}`, null, token);

export const createServiceRequest = (payload, token) =>
  api("POST", "/service-requests", payload, token);

export const updateServiceRequestStatus = (requestId, status, token) =>
  api("PUT", `/service-requests/${requestId}/status?status=${status}`, null, token);

File: case_study/Backend/assetmanagement/src/main/java/hexaware/casestudy/assetmanagement/service/ServiceRequestService.java
package hexaware.casestudy.assetmanagement.service;

import hexaware.casestudy.assetmanagement.dto.request.ServiceRequestDto;
import hexaware.casestudy.assetmanagement.dto.response.ServiceRequestResponse;
import hexaware.casestudy.assetmanagement.enums.ServiceStatus;
import java.util.List;

public interface ServiceRequestService {
    ServiceRequestResponse createServiceRequest(ServiceRequestDto request);
    List<ServiceRequestResponse> getAllServiceRequests();
    List<ServiceRequestResponse> getServiceRequestsByEmployee(Long employeeId);
    ServiceRequestResponse updateStatus(Long requestId, ServiceStatus status);
}